From 32073ad0757f5955a8a3395dd07b674af2cd87a0 Mon Sep 17 00:00:00 2001
From: Michael Larabel <Michael@phoronix.com>
Date: Tue, 15 Nov 2016 08:29:32 -0600
Subject: [PATCH 2/3] log_exporter: Allow easily exporting test run and system
 logs via COPY_TEST_RUN_LOGS_TO and COPY_SYSTEM_LOGS_TO env vars

---
 ChangeLog                         |  1 +
 pts-core/modules/log_exporter.php | 67 +++++++++++++++++++++++++++++++++++++++
 pts-core/pts-core.php             |  2 +-
 3 files changed, 69 insertions(+), 1 deletion(-)
 create mode 100644 pts-core/modules/log_exporter.php

diff --git a/ChangeLog b/ChangeLog
index 3be7000..5bcdf69 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -7,6 +7,7 @@ Phoronix Test Suite (Git)
 - pts-core: Add PHP GD check as first thing when attempting to use result-file-to-pdf
 - pts-core: Add new module hooks for __test_log_output() and __post_test_run_system_logs()
 - phodevi: Don't load Phodevi sensors prior to PHP 5.3 due to compatibility issues
+- log_exporter: Allow easily exporting test run and system logs via COPY_TEST_RUN_LOGS_TO and COPY_SYSTEM_LOGS_TO env vars
 - flamegrapher: Experimental new module to generate Linux perf / FlameGraph graphs for each test being executed when active
 - results_custom_export: Example module for dumping PTS test results in custom formats
 - pts_Graph: Add Intel color highlighting for "Celeron" too
diff --git a/pts-core/modules/log_exporter.php b/pts-core/modules/log_exporter.php
new file mode 100644
index 0000000..bc72581
--- /dev/null
+++ b/pts-core/modules/log_exporter.php
@@ -0,0 +1,67 @@
+<?php
+
+/*
+	Phoronix Test Suite
+	URLs: http://www.phoronix.com, http://www.phoronix-test-suite.com/
+	Copyright (C) 2016, Phoronix Media
+	Copyright (C) 2016, Michael Larabel
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+class log_exporter extends pts_module_interface
+{
+	const module_name = 'Log Exporter';
+	const module_version = '1.0.0';
+	const module_description = 'This module allows for easily exporting test run logs and system logs to external locations via specifying the directory paths via the COPY_TEST_RUN_LOGS_TO and COPY_SYSTEM_LOGS_TO environment variables.';
+	const module_author = 'Michael Larabel';
+
+	public static function module_environmental_variables()
+	{
+		return array('COPY_TEST_RUN_LOGS_TO', 'COPY_SYSTEM_LOGS_TO');
+	}
+	public static function __test_log_output($log_file_path)
+	{
+		if(empty(getenv('COPY_TEST_RUN_LOGS_TO')))
+		{
+			return;
+		}
+		$COPY_TEST_RUN_LOGS_TO = getenv('COPY_TEST_RUN_LOGS_TO');
+
+		pts_file_io::mkdir($COPY_TEST_RUN_LOGS_TO);
+		if(is_writable($COPY_TEST_RUN_LOGS_TO))
+		{
+			copy($log_file_path, $COPY_TEST_RUN_LOGS_TO . '/' . basename($log_file_path));
+		}
+	}
+	public static function __post_test_run_system_logs($log_file_path)
+	{
+		if(empty(getenv('COPY_SYSTEM_LOGS_TO')))
+		{
+			return;
+		}
+		$COPY_SYSTEM_LOGS_TO = getenv('COPY_SYSTEM_LOGS_TO');
+
+		pts_file_io::mkdir($COPY_SYSTEM_LOGS_TO);
+		if(is_writable($COPY_SYSTEM_LOGS_TO))
+		{
+			foreach(pts_file_io::glob($log_file_path . '/*') as $sys_log_file)
+			{
+				copy($sys_log_file, $COPY_SYSTEM_LOGS_TO . '/' . basename($sys_log_file));
+			}
+		}
+	}
+}
+
+?>
diff --git a/pts-core/pts-core.php b/pts-core/pts-core.php
index 0bf87d9..0036058 100644
--- a/pts-core/pts-core.php
+++ b/pts-core/pts-core.php
@@ -241,7 +241,7 @@ function pts_version_codenames()
 }
 
 pts_define('PTS_VERSION', '6.8.0m1');
-pts_define('PTS_CORE_VERSION', 6711);
+pts_define('PTS_CORE_VERSION', 6712);
 pts_define('PTS_CODENAME', 'TANA');
 pts_define('PTS_IS_CLIENT', (defined('PTS_MODE') && strstr(PTS_MODE, 'CLIENT') !== false));
 pts_define('PTS_IS_WEB_CLIENT', (defined('PTS_MODE') && PTS_MODE == 'WEB_CLIENT'));
-- 
2.10.1

